src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,354 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+from collections.abc import Generator
+from contextlib import contextmanager
+
+from gt4py import eve
+from gt4py.cartesian import utils
+from gt4py.cartesian.gtc.common import (
+    AxisBound,
+    BuiltInLiteral,
+    CartesianOffset,
+    DataType,
+    FieldAccess,
+    HorizontalInterval,
+    HorizontalMask,
+    LevelMarker,
+    LoopOrder,
+    While,
+)
+from gt4py.cartesian.gtc.definitions import Extent
+from gt4py.cartesian.gtc.oir import (
+    AssignStmt,
+    BinaryOp,
+    CacheDesc,
+    Cast,
+    Decl,
+    FieldDecl,
+    HorizontalExecution,
+    HorizontalRestriction,
+    IJCache,
+    Interval,
+    KCache,
+    Literal,
+    LocalScalar,
+    MaskStmt,
+    NativeFuncCall,
+    ScalarAccess,
+    ScalarDecl,
+    Stencil,
+    Temporary,
+    TernaryOp,
+    UnaryOp,
+    UnboundedInterval,
+    VariableKOffset,
+    VerticalLoop,
+    VerticalLoopSection,
+)
+from gt4py.cartesian.gtc.passes.oir_optimizations.utils import StencilExtentComputer
+from gt4py.eve import codegen
+from gt4py.eve.concepts import SymbolRef
+
+
+class DebugCodeGen(codegen.TemplatedGenerator, eve.VisitorWithSymbolTableTrait):
+    def __init__(self) -> None:
+        self.body = utils.text.TextBlock()
+        self.symbol_table: dict[str, FieldDecl] = {}
+
+    def visit_Stencil(self, stencil: Stencil, **_):
+        self.generate_imports()
+
+        self.generate_run_function(stencil)
+
+        field_extents, block_extents = self.compute_extents(stencil)
+        self.symbol_table = self.initial_declarations(stencil, field_extents)
+        self.generate_stencil_code(stencil, block_extents)
+
+        return self.body.text
+
+    def generate_imports(self):
+        self.body.append("import numpy as np")
+        self.body.append("from gt4py.cartesian.gtc import ufuncs")
+        self.body.append("from gt4py.cartesian.utils import Field")
+
+    @staticmethod
+    def compute_extents(node: Stencil, **_) -> tuple[dict[str, Extent], dict[int, Extent]]:
+        ctx: StencilExtentComputer.Context = StencilExtentComputer().visit(node)
+        return ctx.fields, ctx.blocks
+
+    def initial_declarations(
+        self, stencil: Stencil, field_extents: dict[str, Extent]
+    ) -> dict[str, FieldDecl]:
+        self.body.append("# ===== Domain Description ===== #")
+        self.body.append("i_0, j_0, k_0 = 0,0,0")
```

Comment:
```suggestion
        self.body.append("i_0, j_0, k_0 = 0, 0, 0")
```
just to be consistent with whitespace?